src/app/components/logo/logo.component.ts

```diff
@@ -1,54 +1,146 @@
 import { NgClass } from '@angular/common';
-import { Component, OnInit } from '@angular/core';
-import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
-import { concat, Observable, timer } from 'rxjs';
-import { delay, finalize, map, repeat, startWith, takeWhile, tap } from 'rxjs/operators';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  DestroyRef,
+  computed,
+  effect,
+  inject,
+  signal
+} from '@angular/core';
+
+type RunningSide = 'r' | 'l';
+
+type LogoState = 'running' | 'blinking' | 'rest';
+
+interface RunningState {
+  count: number;
+  side: RunningSide;
+}
 
-@UntilDestroy()
 @Component({
   selector: 't-logo',
   standalone: true,
   imports: [NgClass],
   templateUrl: './logo.component.html',
-  styleUrls: ['./logo.component.scss']
+  styleUrls: ['./logo.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class LogoComponent implements OnInit {
-  className = '';
+export class LogoComponent {
```

Comment:
Let's stick with the old implementation using RxJS. It is easier to read and understand. 
Having the `async` pipe here and there is perfectly fine. 
Personally, I'm not a big fan of using switch cases extensively either.